modules/modelSetup/BaseStableDiffusionXLSetup.py

```diff
@@ -236,11 +319,56 @@ def calculate_loss(
                 args.normalize_masked_area_loss
             ).mean([1, 2, 3])
         else:
-            losses = F.mse_loss(
+            #MSE/L2 Loss
+            l2_losses = F.mse_loss(
                 predicted,
                 target,
                 reduction='none'
             ).mean([1, 2, 3])
+            
+            #MAE/L1 Loss
+            l1_losses = F.l1_loss(
+                predicted, 
+                target, 
+                reduction='none'
+            ).mean([1, 2, 3])
+            
+            #Perceptual Loss (VGG Based)
+            vgg = models.vgg16(weights=models.VGG16_Weights.DEFAULT).features.to('cuda').eval()
+            p_loss = self.perceptual_loss(predicted, target, vgg)
+            p_loss = p_loss / 255
+            
+            #Color Loss (L1 Loss in Lab color space)
```

Comment:
Color loss doesn't make any sense for a latent diffusion model. The latent space doesn't know about colors.